docker/hwaccel-rkmpp.yml

```diff
@@ -0,0 +1,149 @@
+version: "3.8"
+
+# Hardware acceleration for transcoding using RKMPP for Rockchip SOCs
+# This is only needed if you want to use hardware acceleration for transcoding.
+# Supported host OS is Ubuntu Jammy 22.04 with custom ffmpeg from ppa:liujianfeng1994/rockchip-multimedia
+
+services:
+  hwaccel:
+    privileged: true
+    group_add:
+      - video
+    devices:
+      - /dev/rga:/dev/rga
+      - /dev/dri:/dev/dri
+      - /dev/dma_heap:/dev/dma_heap
+      - /dev/mpp_service:/dev/mpp_service
+    volumes:
+      - /lib/firmware/mali_csffw.bin:/lib/firmware/mali_csffw.bin:ro
+      - /usr/bin/ffmpeg:/usr/bin/ffmpeg_mpp:ro
+      - /lib/aarch64-linux-gnu/libavdevice.so.60:/lib/aarch64-linux-gnu/libavdevice.so.60:ro
```

Comment:
do these really need to be enumerated? Could you use a volume of `/lib/aarch64-linux-gnu:/lib/aarch64-linux-gnu` instead?

Reply:
yeah, I tried that first. But it segfaults :-/

Reply:
Oh dang, lol. Even without the read only?

Reply:
yes :-D
I also tried to link the libraries into another folder (so it does never overwrite anything else, e.g. libc in the container) and use `LD_LIBRARY_PATH` - segfault.

Reply:
I just found a way that might work, using multiple `LD_LIBRARY_PATH`.
I'll try to put this into env.. and update the PR if it works

Reply:
found a way - setting `LD_LIBRARY_PATH` dynamically in code is a bit cumbersome as fluent ffmpeg does not support setting environment variables. it works, though.

Reply:
Would it maybe be possible to just add a script that installs this package, without including those packages in the image? So a user could run that script in the container and have it install the necessary packages locally.

Reply:
Installing only works on Ubuntu Jammy. Our base image is debian bookworm.
We could make a script that builds a new local image into which the binaries/libraries are copied. This would need to be run by the user after every docker pull (and is this forgotten)

Reply:
If it's at all possible to keep this scoped in the container, I feel like that's the way to go. We could publish a Jammy version of the server (or maybe switch to Jammy outright?) and gate this package behind a build arg. Then a user could set that flag in their docker-compose to build the image with this package. This would be easy for users to set up, it would avoid mixing host/container dependencies, and we wouldn't be distributing problematic packages ourselves.

Reply:
Or alternatively, we could add a secondary start.sh script that makes sure this package is installed before calling Node.js. That would avoid the need to build the image locally.

Reply:
properly installing the patched ffmpeg has the issue that the normal ffmpeg also uses some of the modified libraries (very undesirable!) unless we do the `LD_LIBRARY_PATH` swapping that is currently implemented. so, there is less benefit in building a custom image.